---
title: "Ath1md"
author: "Lieke Vlaar"
date: "21/05/2021"
output: html_document
---

# Introduction
In this file the complete analysis workflow of count data of the pre-test Arabidopsis RNAseq experiment is shown.

## Goal

The goal of this experiment is to unravel the effect of solanoeclepin A on the plant. Solanoeclepin A is applied on Arabidopsis seedlings under various nutritional conditions. The set up of this experiment is summarized in this figure:
```{r}
xp_design <- read.csv("Ath1/input/xp_design.csv")
xp_design
```

## Setup

Packages necessary:

* Checkpoint
* DESeq2
* dplyr
* tidyr
* tidyverse
* Biostrings
* biomaRt
* biomartr
* clusterProfiler
* org.At.tair.db
* tibble
* mixOmics
* vegan
* apeglm
* EnhancedVolcano
* patchwork

```{r include=FALSE}
#########
# Library
#########
if ("checkpoint" %in% installed.packages()){
  library("checkpoint") # https://cran.r-project.org/web/packages/checkpoint/index.html
} else {
  install.packages("checkpoint")
  suppressPackageStartupMessages(library("checkpoint"))
  suppressPackageStartupMessages(checkpoint("2021-04-20"))  # all packages in your project will be taken from that date.
}
if ("DESeq2" %in% installed.packages()){
  library("DESeq2")
} else {
  BiocManager::install("DESeq2")
  library("DESeq2")
}
if ("dplyr" %in% installed.packages()){
  library("dplyr")
} else {
  install.packages("dplyr")
  library("dplyr")
}
if ("tidyr" %in% installed.packages()){
  library("tidyr")
} else {
  install.packages("tidyr")
  library("tidyr")
}
if ("tidyverse" %in% installed.packages()){
  library("tidyverse")
} else {
  install.packages("tidyverse")
  library("tidyverse")
}
if ("Biostrings" %in% installed.packages()){
  library("Biostrings") 
} else {
  BiocManager::install("Biostrings")
  library("Biostrings")
}
if ("biomaRt" %in% installed.packages()){
  library("biomaRt") 
} else {
  BiocManager::install("biomaRt")
  library("biomaRt")
}
if ("biomartr" %in% installed.packages()){
  library("biomartr")
} else {
  install.packages("biomartr", dependencies = TRUE)
  library("biomartr")
}
if ("clusterProfiler" %in% installed.packages()){
  library("clusterProfiler") 
} else {
  BiocManager::install("clusterProfiler")
  library("clusterProfiler")
}
if ("org.At.tair.db" %in% installed.packages()){
  library("org.At.tair.db") 
} else {
  BiocManager::install("org.At.tair.db")
  library("org.At.tair.db")
}
if ("tibble" %in% installed.packages()){
  library("tibble") 
} else {
  install.packages("tibble")
  library("tibble")
}
if ("mixOmics" %in% installed.packages()){
  library("mixOmics") 
} else {
  BiocManager::install("mixOmics")
  library("mixOmics")
}
if ("vegan" %in% installed.packages()){
  library("vegan") 
} else {
  install.packages("vegan")
  library("vegan")
}
if ("apeglm" %in% installed.packages()){
  library("apeglm") 
} else {
  BiocManager::install("apeglm")
  library("apeglm")
}
if ("EnhancedVolcano" %in% installed.packages()){
  library("EnhancedVolcano") 
} else {
  BiocManager::install("EnhancedVolcano")
  library("EnhancedVolcano")
}
if ("patchwork" %in% installed.packages()){
  library("patchwork") 
} else {
  BiocManager::install("patchwork")
  library("patchwork")
}
if ("reshape" %in% installed.packages()){
  library("reshape") 
} else {
  install.packages("reshape")
  library("reshape")
}
if ("pheatmap" %in% installed.packages()){
  library("pheatmap") 
} else {
  BiocManager::install("pheatmap")
  library("pheatmap")
}
```

# Mapping results

Calculate the average unique mapping %:

```{r}
ms <- read.csv("Ath1/input/mapping_summary.csv")
ms <- ms[, -1] %>% column_to_rownames("attribute")
ms[5,3]
```

```{r include = FALSE}
unique <- ms[9,]
unique <- as.list(unique)
unique <- gsub("%", "", unique)
unique <- unique[-1]
unique <- as.numeric(unique)
uniquepercent <- mean(unique)
```
```{r}
uniquepercent
```

The average unique mapping % is 89.91%.




# PCA analysis

Load the function:
```{r}
source("Ath1/scripts/create_pca_plot.R")
```

Run the function with default settings, which is
```{r, eval = FALSE}
plot_pca <- function(count_csv_file = "Ath1/input/counts.csv",
                     xp_design_csv_file = "Ath1/input/xp_design.csv",
                     pc_x_axis = "PC1", 
                     pc_y_axis = "PC2",
                     timepoint = c("2","6","24"),
                     pca_colour = "treatment")
```

```{r}
plot_pca()
```

In order to make an overview of what determines the clustering, make the same plot with different colour guides and group them together with patchwork package
```{r}
p1 <- plot_pca(pca_colour = "treatment")
p2 <- plot_pca(pca_colour = "time")
p3 <- plot_pca(pca_colour = "sequencing_batch")
p4 <- plot_pca(pca_colour = "rna_isolation_batch")
p5 <- plot_pca(pca_colour = "purification_procedure")

p1 + p2 + p3 + p4 + p5 +plot_layout(ncol = 2)
```

From these plots, it is clear that PC2 is explaining the time, but PC1 shows unexplained variation and also the solA treatment is not visible.
Let's explore more components to see if we can see the solA treatment in the plots.
```{r}
p1 <- plot_pca(pc_x_axis = "PC3", pc_y_axis = "PC4", pca_colour = "treatment")
p2 <- plot_pca(pc_x_axis = "PC3", pc_y_axis = "PC4", pca_colour = "time")
p3 <- plot_pca(pc_x_axis = "PC3", pc_y_axis = "PC4", pca_colour = "sequencing_batch")
p4 <- plot_pca(pc_x_axis = "PC3", pc_y_axis = "PC4", pca_colour = "rna_isolation_batch")
p5 <- plot_pca(pc_x_axis = "PC3", pc_y_axis = "PC4", pca_colour = "purification_procedure")

p1 + p2 + p3 + p4 + p5 +plot_layout(ncol = 2)
```

```{r}
p1 <- plot_pca(pc_x_axis = "PC5", pc_y_axis = "PC6", pca_colour = "treatment")
p2 <- plot_pca(pc_x_axis = "PC5", pc_y_axis = "PC6", pca_colour = "time")
p3 <- plot_pca(pc_x_axis = "PC5", pc_y_axis = "PC6", pca_colour = "sequencing_batch")
p4 <- plot_pca(pc_x_axis = "PC5", pc_y_axis = "PC6", pca_colour = "rna_isolation_batch")
p5 <- plot_pca(pc_x_axis = "PC5", pc_y_axis = "PC6", pca_colour = "purification_procedure")

p1 + p2 + p3 + p4 + p5 +plot_layout(ncol = 2)
```

Even up to PC6, there is no clear treatment effect. Also batches do not influence the PC1 - PC6. Only time shows a clear effect. This means that there are factors influencing the data that we do not know of.

Check the screeplot of all the treatments.

```{r}
source("Ath1/scripts/scree_plot.R")
plot_scree()
```

```{r}
source("Ath1/scripts/produce_scaled_counts_matrix.R")
scaled_counts <- produce_scaled_counts_matrix()

source("Ath1/scripts/filter_counts_based_on_timepoint.R")
f <- filter_counts_based_on_timepoint()
ft <- f %>% t() %>% as.data.frame() %>% gather ()
ggplot(ft, aes(value)) + 
  geom_histogram(bins = 10) + 
  facet_wrap(~key, scales = 'free_x') +
  scale_y_log10()

f_no_0 <- f[vapply(f, 
                   function(z) length(unique(z)) > 1,
                   logical(1L))] %>% data.matrix(., rownames.force = T)

median_all_genes <- median(f_no_0)
median_per_gene <- apply(f_no_0, 2, median)
f_no_0_t <- f_no_0 %>% t() %>% as.data.frame()
f_f <- f_no_0_t %>% filter(median_per_gene > median_all_genes) %>% t() %>% as.data.frame()

f_ft <- f_f %>% t() %>% as.data.frame() %>% gather()
head(f_ft)
ggplot(f_ft, aes(value)) + 
  geom_histogram(bins = 10) + 
  facet_wrap(~key, scales = 'free_x') +
  scale_y_log10() 

```

However, this experiment is based on time point specific comparisons. We have different timepoints (2, 6 and 24) and within these conditions we compare the effect of adding solA. There could be an effect of solA in one condition, but not in the others. Therefore, it makes sense to look whether samples within conditions cluster based on solA treatment.
```{r}
p1 <- plot_pca(pc_x_axis = "PC1", pc_y_axis = "PC2", tp = 2) + ggtitle("t=2h")
p2 <- plot_pca(pc_x_axis = "PC1", pc_y_axis = "PC2", tp = 6) + ggtitle("t=6h")
p3 <- plot_pca(pc_x_axis = "PC1", pc_y_axis = "PC2", tp = 24) + ggtitle("t=24h")

p1 + p2 + p3 + plot_layout(ncol = 2)
```

Not much of solanoeclepin A effect. Let's check the other components.
```{r}
p1 <- plot_pca(pc_x_axis = "PC3", pc_y_axis = "PC4", tp = 2) + ggtitle("t=2h")
p2 <- plot_pca(pc_x_axis = "PC3", pc_y_axis = "PC4", tp = 6) + ggtitle("t=6h")
p3 <- plot_pca(pc_x_axis = "PC3", pc_y_axis = "PC4", tp = 24) + ggtitle("t=24h")

p1 + p2 + p3 + plot_layout(ncol = 2)

p1 <- plot_pca(pc_x_axis = "PC5", pc_y_axis = "PC6", tp = 2) + ggtitle("t=2h")
p2 <- plot_pca(pc_x_axis = "PC5", pc_y_axis = "PC6", tp = 6) + ggtitle("t=6h")
p3 <- plot_pca(pc_x_axis = "PC5", pc_y_axis = "PC6", tp = 24) + ggtitle("t=24h")

p1 + p2 + p3 + plot_layout(ncol = 2)
```

Up to PC6, there is no clear solanoeclepin A effect. Other unknown factors are influencing our data more.

# DEG analysis

## Wald test

Use design formula ~ time + treatment
```{r message=FALSE, warning=FALSE}
source("Ath1/scripts/volcanoplot.R")
```
```{r, warning = FALSE, message = FALSE}
make_volcanoplot(log2FC_threshold = 0, FCcutoff_volcano = 1, padj_threshold = 0.01, ttl = "Micromolar solA, all timepoints, Wald", method = "time+treatment")
make_volcanoplot(trtm = c("ethanol", "nanomolar_solanoeclepinA"), treatment2 = "nanomolar_solanoeclepinA", log2FC_threshold = 0, FCcutoff_volcano = 1, padj_threshold = 0.01, ttl = "Nanomolar solA, all timepoints, Wald", method = "time+treatment")
```

So especially in the nM treatment, there are some DEGs, taking all timepoints into account. For the micromolar treatment, there is only 1.
Let's see if it overlaps:

``` {r message = FALSE, warning = FALSE}
source("Ath1/scripts/get_filtered_list_of_DEGs.R")
```
```{r message = FALSE, warning = FALSE}
umolar <- get_list_of_DEGs() %>% rownames_to_column("Gene")
nmolar <- get_list_of_DEGs(trtm = c("ethanol", "nanomolar_solanoeclepinA"), treatment2 = "nanomolar_solanoeclepinA") %>% rownames_to_column("Gene")
inner_join(umolar, nmolar, by = "Gene")
```

Indeed, that gene, AT5G56550.1, overlaps with the nanomolar treatment.

We can also check if there are DEGs specifically for a timepoint.
First time point = 2h:
```{r message = FALSE, warning = FALSE}
make_volcanoplot(tp = 2, method = "treatment", FCcutoff_volcano = 1, padj_threshold = 0.01, ttl = "DEGs at t = 2h, uM solA, Wald test")
make_volcanoplot(trtm = c("ethanol", "nanomolar_solanoeclepinA"), treatment2 = "nanomolar_solanoeclepinA", tp = 2, method = "treatment", FCcutoff_volcano = 1, padj_threshold = 0.01, ttl = "DEGs at t = 2h, nM solA, Wald test")
```

Then for time point = 6h:
```{r message = FALSE, warning = FALSE}
make_volcanoplot(tp = 6, method = "treatment", FCcutoff_volcano = 1, padj_threshold = 0.01, ttl = "DEGs at t = 6h, uM solA, Wald test")
make_volcanoplot(trtm = c("ethanol", "nanomolar_solanoeclepinA"), treatment2 = "nanomolar_solanoeclepinA", tp = 6, method = "treatment", FCcutoff_volcano = 1, padj_threshold = 0.01, ttl = "DEGs at t = 6h, nM solA, Wald test")
```

Then for time point = 24h:
```{r message = FALSE, warning = FALSE}
make_volcanoplot(tp = 24, method = "treatment", FCcutoff_volcano = 1, padj_threshold = 0.01, ttl = "DEGs at t = 24h, uM solA, Wald test")
make_volcanoplot(trtm = c("ethanol", "nanomolar_solanoeclepinA"), treatment2 = "nanomolar_solanoeclepinA", tp = 24, method = "treatment", FCcutoff_volcano = 1, padj_threshold = 0.01, ttl = "DEGs at t = 24h, nM solA, Wald test")
```


## LRT

Compare Wald test with LRT test, which checks the null-model with the full model. This is a particularly suitable method for experiments with multiple time points.
First for the uM solA treatment:
```{r message=FALSE, warning=FALSE}
source("Ath1/scripts/compare_Wald_vs_LRT.R")
u <- compare_wald_vs_LRT()
length(row.names(u))
u
```

Then for the nM solA treatment:
```{r message=FALSE, warning=FALSE}
n <- compare_wald_vs_LRT(trtm = c("ethanol", "nanomolar_solanoeclepinA"),
                    treatment2 = "nanomolar_solanoeclepinA")
length(row.names(n))
n
```

Do the same for the separate time points to see if there are any confirmed DEGs:
t=2h
```{r message = FALSE, warning=FALSE}
u2 <- compare_wald_vs_LRT(tp = 2, method = "treatment")
n2 <- compare_wald_vs_LRT(tp = 2, method = "treatment", trtm = c("ethanol", "nanomolar_solanoeclepinA"),
                    treatment2 = "nanomolar_solanoeclepinA")
length(row.names(u2))
length(row.names(n2))
```

Then for t=6h
```{r message=FALSE, warning=FALSE}
u6 <- compare_wald_vs_LRT(tp = 6, method = "treatment")
n6 <- compare_wald_vs_LRT(tp = 6, method = "treatment", trtm = c("ethanol", "nanomolar_solanoeclepinA"),
                    treatment2 = "nanomolar_solanoeclepinA")
length(row.names(u6))
length(row.names(n6))
```

And for t = 24h:
```{r message=FALSE, warning = FALSE}
u24 <- compare_wald_vs_LRT(tp = 24, method = "treatment")
n24 <- compare_wald_vs_LRT(tp = 24, method = "treatment", trtm = c("ethanol", "nanomolar_solanoeclepinA"),
                    treatment2 = "nanomolar_solanoeclepinA")
length(row.names(u24))
length(row.names(n24))
```

From these results, it seems that solanoeclepin A treatment is mostly affecting gene expression at 2h and 24h, and at nanomolar concentrations. Let's visualise this in a barplot:
```{r message = FALSE, warning = FALSE}
samples <- c("all samples, uM solA", "all samples, nM solA","t=2h, uM solA" ,"t=2h, nM solA","t=6h, uM solA","t=6h, nM solA", "t=24h, uM solA", "t=24h, nM solA")
overlap <- c(length(row.names(u)), length(row.names(n)), length(row.names(u2)), length(row.names(n2)), length(row.names(u6)), length(row.names(n6)), length(row.names(u24)), length(row.names(n24)))
x <- data.frame(samples, overlap)
x <- melt(x)
ggplot(data = x,
       aes(x = samples, y = value, fill = variable)) +
  geom_bar(position = "dodge",
           stat = "identity") +
  ggtitle("DEGs by solA treatment, different sample subgroups") +
  xlab("") +
  ylab("# DEGs") + 
  theme_minimal() +
  scale_fill_manual(values = c("goldenrod3","sienna3","dodgerblue3")) +
  coord_flip() +
  geom_text(aes(label=value), 
            position = position_dodge(width = 0.9), 
            vjust = 0.3, 
            hjust = 1.2,
            color="white", 
            size=3.5)


```

Let's see if the DEGs from 2 and 24h are overlapping, or if these are different genes.

```{r message=FALSE, warning=FALSE}
inner_join(n2, n24, by = "gene")
```

Only 13 genes overlap between these timepoints. This means that a nanomolar solanoeclepin A treatment induces mostly timepoint-specific transcriptional changes.

## Venn diagram

```{r message = FALSE, warning = FALSE}
x1 <- list(n2$gene,u2$gene,n6$gene,n24$gene,u24$gene)
v <- venn.diagram(x1,
             category.names = c(paste0("nM, 2h (", nrow(n2), ")"),
                                paste0("uM, 2h (", nrow(u2), ")"), 
                                paste0("nM, 6h (", nrow(n6), ")"),
                                paste0("nM, 24h (", nrow(n24), ")"),
                                paste0("uM, 24h (", nrow(u24), ")")),
             col=c("aquamarine","aquamarine4", "darkorange2","magenta","maroon4"),
             fill = c(alpha("aquamarine",0.3), alpha('aquamarine4',0.3), alpha('darkorange2',0.3), alpha("magenta", 0.3), alpha("maroon4", 0.3)),
             filename = "Ath1/output/Ath1_venn.png",
             output = TRUE,
             cex = 1.5,
             cat.cex = 1,
             cat.default.pos = "text",
             cat.pos = c(0, -20, -140, 140, -20),
             cat.fontfamily = "sans",
             fontfamily = "sans",
             ext.text = TRUE)

```

Conclusion from the Venn diagram is that there is very little overlap between DEGs between timepoints and concentrations. 

# Annotation of DEGs

Start with annotating the list of DEGs of each subset. The results will be saved in the Ath1/output/ folder in the format 'annotated_DEGs_[conc_of_treatment]_[timepoint].csv'.
```{r message = FALSE, warning = FALSE}
source("Ath1/scripts/get_annotated_DEGs.R")
DEGs_uM <- get_annotated_DEGs(name = "Ath1/output/annotated_DEGs_uM_alltp.csv")
DEGs_nM <- get_annotated_DEGs(trtm = c("ethanol", "nanomolar_solanoeclepinA"), treatment2 = "nanomolar_solanoeclepinA", name = "Ath1/output/annotated_DEGs_nM_alltp.csv")
DEGs_uM2 <- get_annotated_DEGs(tp = 2, method = "treatment", name = "Ath1/output/annotated_DEGs_uM_tp2.csv")
DEGs_nM2 <- get_annotated_DEGs(tp = 2, trtm = c("ethanol", "nanomolar_solanoeclepinA"), treatment2 = "nanomolar_solanoeclepinA", method = "treatment", name = "Ath1/output/annotated_DEGs_nM_tp2.csv")
DEGs_nM6 <- get_annotated_DEGs(tp = 6, trtm = c("ethanol", "nanomolar_solanoeclepinA"), treatment2 = "nanomolar_solanoeclepinA", method = "treatment", name = "Ath1/output/annotated_DEGs_nM_tp6.csv")
DEGs_uM24 <- get_annotated_DEGs(tp = 24, method = "treatment", name = "Ath1/output/annotated_DEGs_uM_tp24.csv")
DEGs_nM24 <- get_annotated_DEGs(tp = 24, trtm = c("ethanol", "nanomolar_solanoeclepinA"), treatment2 = "nanomolar_solanoeclepinA", method = "treatment", name = "Ath1/output/annotated_DEGs_nM_tp24.csv")
```

# Visualisation of DEG analysis

We will do heatmaps of DEGs and GO enrichment analysis

## Heatmap

From the list of DEGs that we compiled above, we will summarize this information into a heatmap.
Let's first look at the timepoint with the most DEGs, 24h.
For this heatmap we use all the DEG from both DEGs_uM24 and DEGs_nM24.

```{r message=FALSE, warning=FALSE}
x2 <- DEGs_nM24
x1 <- DEGs_uM24
x <- rbind(x1,x2)
source("Ath1/scripts/produce_scaled_counts_matrix.R")
y <- produce_scaled_counts_matrix() %>% column_to_rownames("sample") %>% t() %>% as.data.frame() %>% rownames_to_column("fullgene")
DEG_counts <- left_join(x, y, by="fullgene") 
DEG_counts <- DEG_counts[, -c(1:8,10:15)]
DEG_counts <- distinct(DEG_counts, fullgene, .keep_all = TRUE)
DEG_counts <- DEG_counts %>% column_to_rownames("fullgene")
DEG_counts <- na.omit(DEG_counts)
DEG_counts <- DEG_counts[,c(12:16,27:31,42:46)]

require(RColorBrewer)
myCol <- colorRampPalette(c("red1", "black", "dodgerblue"))(100)
myBreaks <- seq(-2, 2, length.out=100)

clrs <- list(
  solA = c(no = "gray", uM ="seagreen4", nM = "mediumspringgreen"))

solA <- rep(c("no", "uM", "nM"), c(5,5,5))
sample <- names(DEG_counts)
sip <- data.frame(sample, solA)%>% column_to_rownames("sample")

pheatmap(DEG_counts,
         gaps_col = c(5,10),
         show_rownames = FALSE,
         show_colnames = FALSE,
         cluster_rows = TRUE,
         cluster_cols = FALSE,
         scale = "row",
         col = myCol,
         treeheight_row = 0,
         cutree_rows = 2,
         annotation_col = sip,
         annotation_colors = clrs)
```

Let's also make a heatmap for t=2

```{r message=FALSE, warning=FALSE}
x1 <- DEGs_nM2
x2 <- DEGs_uM2
x <- rbind(x1, x2)
DEG_counts <- left_join(x, y, by="fullgene") 
DEG_counts <- DEG_counts[, -c(1:8,10:15)]
DEG_counts <- distinct(DEG_counts, fullgene, .keep_all = TRUE)
DEG_counts <- DEG_counts %>% column_to_rownames("fullgene")
DEG_counts <- na.omit(DEG_counts)
DEG_counts <- DEG_counts[,c(1:6,17:21,32:36)]

solA <- rep(c("no", "uM", "nM"), c(6,5,5))
sample <- names(DEG_counts)
sip <- data.frame(sample, solA)%>% column_to_rownames("sample")

pheatmap(DEG_counts,
         gaps_col = c(6,11),
         show_rownames = FALSE,
         show_colnames = FALSE,
         cluster_rows = TRUE,
         cluster_cols = FALSE,
         scale = "row",
         col = myCol,
         treeheight_row = 0,
         cutree_rows = 2,
         annotation_col = sip,
         annotation_colors = clrs)
```


```{r message=FALSE, warning=FALSE}
x <- DEGs_nM6
DEG_counts <- left_join(x, y, by="fullgene") 
DEG_counts <- DEG_counts[, -c(1:8,10:15)]
DEG_counts <- distinct(DEG_counts, fullgene, .keep_all = TRUE)
DEG_counts <- DEG_counts %>% column_to_rownames("fullgene")
DEG_counts <- na.omit(DEG_counts)
DEG_counts <- DEG_counts[,c(7:11,22:26,37:41)]

solA <- rep(c("no", "uM", "nM"), c(5,5,5))
sample <- names(DEG_counts)
sip <- data.frame(sample, solA)%>% column_to_rownames("sample")

pheatmap(DEG_counts,
         gaps_col = c(5,10),
         show_rownames = FALSE,
         show_colnames = FALSE,
         cluster_rows = TRUE,
         cluster_cols = FALSE,
         scale = "row",
         col = myCol,
         treeheight_row = 0,
         cutree_rows = 2,
         annotation_col = sip,
         annotation_colors = clrs,
         border_color = NA)
```
From the heatmap it is clear that the uM treatment is not so different from the mock treatment.

We can summarize this in one big figure with the DEGs found when taking the complete dataset into account.

```{r message=FALSE, warning=FALSE}
x1 <- DEGs_nM
x2 <- DEGs_uM
x <- rbind(x1, x2)
DEG_counts <- left_join(x, y, by="fullgene") 
DEG_counts <- DEG_counts[, -c(1:8,10:15)]
DEG_counts <- distinct(DEG_counts, fullgene, .keep_all = TRUE)
DEG_counts <- DEG_counts %>% column_to_rownames("fullgene")
DEG_counts <- na.omit(DEG_counts)

solA <- rep(c("no", "uM", "nM","no","uM","nM","no","uM","nM"), c(6,5,5,5,5,5,5,5,5))
time <- rep(c("2","6","24"), c(16,15,15))
sample <- names(DEG_counts)
sip <- data.frame(sample, solA, time)%>% column_to_rownames("sample")

clrs <- list(
  solA = c(no = "gray", uM ="seagreen4", nM = "mediumspringgreen"),
  time = c("2" = "yellow", "6" = "darkgoldenrod1", "24" = "darkorange"))

pheatmap(DEG_counts,
         gaps_col = c(16,31),
         show_rownames = FALSE,
         show_colnames = FALSE,
         cluster_rows = TRUE,
         cluster_cols = FALSE,
         scale = "row",
         col = myCol,
         treeheight_row = 0,
         cutree_rows = 2,
         annotation_col = sip,
         annotation_colors = clrs)
```


## GO annotation 

GO annotation can summarize the function of DEGs and thereby give an overview of the effects of a treatment. GO enrichment can be summarized in a dotplot, like this:

```{r message = FALSE, warning = FALSE}
source("Ath1/scripts/dotplot.R")
plot_dot()
```

In order to compare samples, we can also depict multiple GO enrichment analyses in one plot. Let's combine all of them:
```{r message = FALSE, warning = FALSE}
source("Ath1/scripts/summ_dotplot.R")
sumplotdot()
```
This is hardly a nice overview because there are so many GO terms enriched. Let's make the FDR more stringent to reduce the number of terms. Default is 0.05, we set it to 0.001. Also, let's only look at one timepoint, 24h:

```{r message = FALSE, warning = FALSE}
sumplotdot(time = "tp24", updown = "all", sign = 0.001)
```

Pay attention to the terms that are present in both samples, such as 'response to chitin', 'biological process involved in interspecies interaction between organisms', 'response to external biotic stimulus', 'response to other organism' and 'response to stress'.
We can also look only at upregulated genes (the downregulated genes of this dataset do not render any enriched GO terms). Let's see which GO terms that gives:

```{r message = FALSE, warning = FALSE}
sumplotdot(time = "tp24", updown = "up", sign = 0.001)
```

When we look only at upregulated GO terms, we see that, besides the GO terms related to biotic interaction we already saw above, now common terms are also related to 'cellular response to hypoxia'. 
The other 2 timepoints, 2h and 6h after solA treatment, only show enriched GO terms in the nM treatment. Let's see what GO terms are enriched in those samples. Plotting with significance level 0.001 or 0.01 does not give any enriched GO terms, so we use the default of 0.05. The uM treatment does not result in any GO terms either, so we will see only nM

```{r message = FALSE, warning = FALSE}
sumplotdot(time = c("tp2","tp6"), updown = c("up", "all"))
```
In this plot the colours are also more purple than red, which means higher (less significant) P values.
The GO terms do not overlap at all. At t=2h, there are mostly enriched terms related to metabolic and biosynthetic processes, whereas at t = 6h, we see again the response to hypoxia.

In conclusion, we can say that a strong response to solanoeclepin A is mostly seen after 24h, when we see a lot of GO terms enriched that are related to biotic interactions. We see this for both concentrations of solanoeclepin A. At 2h and 6h after treatment, we see demarcated responses which are related to metabolic and biosynthetic processes and hypoxia response, respectively.





