---
title: "Sly"
author: "Lieke Vlaar"
date: "30/04/2021"
output: html_document
---
# Introduction
In this file the complete analysis workflow of count data of the tomato RNAseq experiment is shown.

## Goal

The goal of this experiment is to unravel the effect of solanoeclepin A on the plant. Solanoeclepin A is applied on tomato seedlings and roots and shoots are harvested separately. The set up of this experiment is summarized in this figure:
```{r include = FALSE}
if ("knitr" %in% installed.packages()){
  library("knitr")
} else {
  install.packages("knitr")
  library("knitr")
}
```
```{r setup, include = FALSE}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
```
```{r figurename, echo=FALSE, out.width = '90%'}
knitr::include_graphics("Sly1/image.png")
```

## Setup

Packages necessary:

* Checkpoint
* DESeq2
* dplyr
* tidyr
* tidyverse
* Biostrings
* biomaRt
* biomartr
* clusterProfiler
* tibble
* mixOmics
* vegan
* apeglm
* EnhancedVolcano
* patchwork
* GenomicFeatures

```{r include=FALSE}
#########
# Library
#########
if ("checkpoint" %in% installed.packages()){
  library("checkpoint") # https://cran.r-project.org/web/packages/checkpoint/index.html
} else {
  install.packages("checkpoint")
  suppressPackageStartupMessages(library("checkpoint"))
  suppressPackageStartupMessages(checkpoint("2021-04-20"))  # all packages in your project will be taken from that date.
}
if ("DESeq2" %in% installed.packages()){
  library("DESeq2")
} else {
  BiocManager::install("DESeq2")
  library("DESeq2")
}
if ("dplyr" %in% installed.packages()){
  library("dplyr")
} else {
  install.packages("dplyr")
  library("dplyr")
}
if ("tidyr" %in% installed.packages()){
  library("tidyr")
} else {
  install.packages("tidyr")
  library("tidyr")
}
if ("tidyverse" %in% installed.packages()){
  library("tidyverse")
} else {
  install.packages("tidyverse")
  library("tidyverse")
}
if ("Biostrings" %in% installed.packages()){
  library("Biostrings") 
} else {
  BiocManager::install("Biostrings")
  library("Biostrings")
}
if ("biomaRt" %in% installed.packages()){
  library("biomaRt") 
} else {
  BiocManager::install("biomaRt")
  library("biomaRt")
}
if ("biomartr" %in% installed.packages()){
  library("biomartr")
} else {
  install.packages("biomartr", dependencies = TRUE)
  library("biomartr")
}
if ("clusterProfiler" %in% installed.packages()){
  library("clusterProfiler") 
} else {
  BiocManager::install("clusterProfiler")
  library("clusterProfiler")
}
if ("tibble" %in% installed.packages()){
  library("tibble") 
} else {
  install.packages("tibble")
  library("tibble")
}
if ("mixOmics" %in% installed.packages()){
  library("mixOmics") 
} else {
  BiocManager::install("mixOmics")
  library("mixOmics")
}
if ("vegan" %in% installed.packages()){
  library("vegan") 
} else {
  install.packages("vegan")
  library("vegan")
}
if ("apeglm" %in% installed.packages()){
  library("apeglm") 
} else {
  BiocManager::install("apeglm")
  library("apeglm")
}
if ("EnhancedVolcano" %in% installed.packages()){
  library("EnhancedVolcano") 
} else {
  BiocManager::install("EnhancedVolcano")
  library("EnhancedVolcano")
}
if ("patchwork" %in% installed.packages()){
  library("patchwork") 
} else {
  BiocManager::install("patchwork")
  library("patchwork")
}
if ("reshape" %in% installed.packages()){
  library("reshape") 
} else {
  install.packages("reshape")
  library("reshape")
}
if ("pheatmap" %in% installed.packages()){
  library("pheatmap") 
} else {
  BiocManager::install("pheatmap")
  library("pheatmap")
}
if ("GenomicFeatures" %in% installed.packages()){
  library("GenomicFeatures") 
} else {
  BiocManager::install("GenomicFeatures")
  library("GenomicFeatures")
}
if ("patchwork" %in% installed.packages()){
  library("patchwork") 
} else {
  BiocManager::install("patchwork")
  library("patchwork")
}
```

# Mapping results

Calculate the average unique mapping %:

```{r}
ms <- read.csv("Sly1/input/mapping_summary.csv")
ms <- ms[-c(37,38), -1] %>% column_to_rownames("attribute")
ms[1:10,1:3]
```

```{r include = FALSE}
unique <- ms[9,]
unique <- as.list(unique)
unique <- gsub("%", "", unique)
unique <- as.numeric(unique)
uniquepercent <- mean(unique)
```
```{r}
uniquepercent
```

The average unique mapping % is 54.0%. This is low, but explained by the type of library prep method that was used (Moll et al, 2014).


# PCA analysis

Plot PCA to see if samples with the same treatments cluster together.

Figure S1A
```{r message=FALSE, warning=FALSE}
source("Sly1/scripts/create_pca_plot.R")
p <- plot_pca(plantpart = c("root", "shoot"), pca_colour = "compartment", pca_shape = "treatment")
p + scale_shape_discrete(name = "treatment", labels = c("control", "solA")) +
  scale_color_manual(values=c('#0099FF','#E69F00'))
```

On PC1, the compartments cluster together. There is no clear clustering based on solA treatment. Let's check PC3 and PC4.
```{r message=FALSE, warning=FALSE}
p1 <- plot_pca(plantpart = c("root", "shoot"), pc_x_axis = "PC3", pc_y_axis = "PC4", pca_colour = "compartment", pca_shape = "treatment")

p1 + scale_color_manual(values=c('#e956b4', '#56e98b'), name = "solanoeclepin A", labels = c("no", "yes")) + guides(shape = FALSE)
```

Also no good separation. Let's plot the screeplot:
```{r message=FALSE, warning=FALSE}
source("Sly1/scripts/scree_plot.R")
plot_scree(comp = c("root","shoot"))
```

It would be more informative to look at the compartments individually.

Figure S1B
```{r}
p <- plot_pca(plantpart = "root", pc_x_axis = "PC1", pc_y_axis = "PC5", pca_shape = "treatment")
p + scale_color_manual(values=c('#e956b4', '#56e98b'), name = "treatment", labels = c("control", "solA")) + guides(shape = FALSE)
```

In root, there seems to be solA separation in PC5, in shoot, clustering based on solA treatment is difficult to observe. But since the treatment is applied on the roots, this might not be so surprising.

Fig S1C

```{r}
p <- plot_pca(plantpart = "shoot", pc_x_axis = "PC1", pc_y_axis = "PC2", pca_shape = "treatment")
p + scale_color_manual(values=c('#566be9', '#d71c37'), name = "treatment", labels = c("control", "solA")) + guides(shape = FALSE)
```



Check the screeplots for separate compartment PCA:
```{r message=FALSE, warning=FALSE}
p1 <- plot_scree(comp = "shoot") + ggtitle("Shoot")
p2 <- plot_scree(comp = "root") + ggtitle("Root")
p1 + p2
```

# DEG analysis

## Wald test

```{r message=FALSE, warning=FALSE}
source("Sly1/scripts/volcanoplot.R")
```


Again, it makes more sense to look within compartments.
```{r, fig.height=12, fig.width=5, warning = FALSE, message = FALSE}
p1 <- make_volcanoplot(plantpart = c("shoot"), log2FC_threshold = 0, FCcutoff_volcano = 0, padj_threshold = 0.05, ttl = "Shoot, solA treatment, Wald", method = "treatment")
p2 <- make_volcanoplot(plantpart = c("root"), log2FC_threshold = 0, FCcutoff_volcano = 0, padj_threshold = 0.05, ttl = "Root, solA treatment, Wald", method = "treatment")
p1 / p2
```

Check the root volcano plot; it looks strange.

```{r check_root_volcano, message = FALSE, warning = FALSE}
ddsr <- get_DESeq_dds(plantpart = "root", method = "treatment")
resr <- get_unfiltered_res_dds(plantpart = "root", method = "treatment")
resrd <- as.data.frame(resr)
shrunk <- lfcShrink(dds = dds,
                    res = resr,
                    type = "apeglm",
                    coef = resultsNames(dds)[2])
shrunk <- as.data.frame(shrunk)
ggplot(data=shrunk, aes(x=log2FoldChange, y=-log10(padj))) + geom_point() + xlim(c(-10,10)) + theme_classic() +
    geom_hline(yintercept=-log10(0.05), col="red")
```

The strange shape of the volcano plot is due to the shrinkage of the numerous genes, that had a adjusted p-value of nearly 1. 

Plot the pvalues in a histogram to see their distribution.

```{r pvalue_hist, message = FALSE, warning = FALSE}
use <- resr$baseMean > metadata(resr)$filterThreshold
h1 <- hist(resr$pvalue[!use], breaks=0:50/50, plot=FALSE)
h2 <- hist(resr$pvalue[use], breaks=0:50/50, plot=FALSE)
colori <- c(`do not pass`="khaki", `pass`="powderblue")
barplot(height = rbind(h1$counts, h2$counts), beside = FALSE,
        col = colori, space = 0, main = "", ylab="frequency")
text(x = c(0, length(h1$counts)), y = 0, label = paste(c(0,1)),
     adj = c(0.5,1.7), xpd=NA)
legend("topright", fill=rev(colori), legend=rev(names(colori)))

```

Check if there is a big difference if extracting results with contrast() except for filtering dataset before inputting in dds.

```{r using_contrast, message = FALSE, warning = FALSE}
counts3 <- read.csv(file = "Sly1/input/counts.csv", 
                     header = TRUE, 
                     stringsAsFactors = FALSE,
                     fileEncoding = "UTF-8-BOM") %>% column_to_rownames("Geneid")

  
xp_design3 <- read.csv(file = "Sly1/input/xp_design.csv", 
                       header = TRUE, 
                       stringsAsFactors = FALSE,
                       fileEncoding = "UTF-8-BOM") %>% column_to_rownames("sample")

xp_design3$group <- factor(paste0(xp_design3$compartment, "_", xp_design3$treatment))

dds3 <- DESeqDataSetFromMatrix(countData = counts3,
                                  colData = xp_design3,
                                  design = ~ group)
keep <- rowSums(counts(dds3)) >= 10
dds3 <- dds3[keep,]
dds3 <- DESeq(dds3)
resr3 <- results(dds3, contrast = c("group", "root_yes_solA","root_no_solA")) 
summary(resr3)
use <- resr3$baseMean > metadata(resr3)$filterThreshold
h1 <- hist(resr3$pvalue[!use], breaks=0:50/50, plot=FALSE)
h2 <- hist(resr3$pvalue[use], breaks=0:50/50, plot=FALSE)
colori <- c(`do not pass`="khaki", `pass`="powderblue")
barplot(height = rbind(h1$counts, h2$counts), beside = FALSE,
        col = colori, space = 0, main = "", ylab="frequency")
text(x = c(0, length(h1$counts)), y = 0, label = paste(c(0,1)),
     adj = c(0.5,1.7), xpd=NA)
legend("topright", fill=rev(colori), legend=rev(names(colori)))
resultsNames(dds3)
shrunk <- lfcShrink(dds = dds3,
                    res = resr3,
                    type = "apeglm",
                    coef = resultsNames(dds3)[2])
shrunk <- as.data.frame(shrunk)
ggplot(data=shrunk, aes(x=log2FoldChange, y=-log10(padj))) + geom_point() + xlim(c(-10,10)) + theme_classic() +
    geom_hline(yintercept=-log10(0.05), col="red")
write.csv(shrunk, "Sly1/output/test_contrast_res_shrunk.csv", row.names = TRUE)
#get the shrunk list annotated

res3 <- shrunk %>% rownames_to_column("original_genename")
genes <- res3$original_genename
genes <- substr(genes, start = 6, stop = 21)
res3$gene <- genes
     
annotations <- get_Slycopersicum_annotations()
res3_annotated <- left_join(res3, annotations, by = "gene")
  
uniquepaste <- function(x) { paste(unique(x), sep = ',', collapse = ",") }
res3_annotated <- aggregate(res3_annotated, by = list(res3_annotated$gene), FUN = uniquepaste)
res3_annotated <- res3_annotated[res3_annotated$padj < 0.05, ]
write.csv(res3_annotated, "Sly1/output/test_contrast_res_shrunk_ann.csv", row.names = TRUE)
```

## Heatmaps

Since there is only 1 DEG if we take into account all samples at the same time, it is not possible to make a heatmap in that way. Therefore, we only use within compartment DEGs, add those together, and produce a heatmap.

```{r message=FALSE, warning=FALSE}
source("Sly1/scripts/get_filtered_list_of_DEGs.R")
xs <- get_list_of_DEGs(plantpart = "shoot", method = "treatment", log2FC_threshold = 0)
xs <- xs %>% rownames_to_column("genes")
xr <- get_list_of_DEGs(plantpart = "root", method = "treatment", log2FC_threshold = 0)
xr <- xr %>% rownames_to_column("genes")
```
So in the shoot there are 36 DEGs and in the root 22 DEGs.

```{r message=FALSE, warning=FALSE}
xrr <- xr$genes 
xss <- xs$genes
length(xss[xss %in% xrr])
```

None of the DEGs of root and shoot overlap; they are all different.

```{r message=FALSE, warning=FALSE}
x <- rbind(xs, xr)
source("Sly1/scripts/produce_scaled_counts_matrix.R")
y <- produce_scaled_counts_matrix() %>% column_to_rownames("sample") %>% t() %>% as.data.frame() %>% rownames_to_column("genes")
DEG_counts <- left_join(x, y, by="genes") 
DEG_counts <- DEG_counts[, -c(2,3,4,5,6)]
DEG_counts <- DEG_counts %>% column_to_rownames("genes")

require(RColorBrewer)
myCol <- colorRampPalette(c("red1", "black", "dodgerblue"))(100)
myBreaks <- seq(-2, 2, length.out=100)

clrs <- list(
  treatment = c(control = "gray", solA ="midnightblue"),
  compartment = c(root = "darkorange4", shoot = "darkgreen"))

treatment <- rep(c("control", "solA"), c(10,10))
compartment <- rep(c("root","shoot","root","shoot"), c(5,5,5,5))
sample <- names(DEG_counts)
sip <- data.frame(sample, treatment, compartment)%>% column_to_rownames("sample")

pheatmap(DEG_counts,
         gaps_col = c(10),
         show_rownames = FALSE,
         show_colnames = FALSE,
         cluster_rows = TRUE,                      
         cluster_cols = TRUE,
         scale = "row",
         col = myCol,
         treeheight_row = 0,
         cutree_rows = 1,
         annotation_col = sip,
         annotation_colors = clrs,
         border_color = NA
         )
```

In this heatmap, samples from the same treatment cluster together, which is good. Only S03 (root, no solA) clusters together with the yes solA. Also, samples from the same compartment cluster together. There is more differences visible between samples the shoot than between samples of root, which is expected since shoot resulted in more DEGs.
Let's run the same script but without clustering, to see all samples in their sample group:

```{r message=FALSE, warning=FALSE}
pheatmap(DEG_counts,
         gaps_col = c(5,10,15),
         show_rownames = FALSE,
         show_colnames = FALSE,
         cluster_rows = TRUE,                      
         cluster_cols = FALSE,
         scale = "row",
         col = myCol,
         treeheight_row = 0,
         cutree_rows = 1,
         annotation_col = sip,
         annotation_colors = clrs,
         border_color = NA
         )
```

Let's look more into the root samples and see whether they cluster together based on only their DEGs.

Figure 1:
```{r message=FALSE, warning=FALSE}
x <- get_list_of_DEGs(plantpart = "root", method = "treatment", log2FC_threshold = 0)
x <- x %>% rownames_to_column("genes")

source("Sly1/scripts/filter_counts_based_on_compartment.R")
y <- filter_counts_based_on_compartment(comprtmt = "root") %>% t() %>% as.data.frame() %>% rownames_to_column("genes")
DEG_counts <- left_join(x, y, by="genes") 
DEG_counts <- DEG_counts[, -c(2,3,4,5,6)]
DEG_counts <- DEG_counts %>% column_to_rownames("genes")
xp_design <- xp_design %>% column_to_rownames("sample")

clrs <- list(
  treatment = c(control= "gray", solA ="midnightblue"))

treatment <- rep(c("control", "solA"), c(5,5))
sample <- names(DEG_counts)
sip <- data.frame(sample, treatment)%>% column_to_rownames("sample")

pheatmap(DEG_counts,
         gaps_col = c(5),
         show_rownames = FALSE,
         show_colnames = FALSE,
         cluster_rows = TRUE,                      
         cluster_cols = TRUE,
         scale = "row",
         col = myCol,
         treeheight_row = 0,
         cutree_rows = 1,
         annotation_col = sip,
         annotation_colors = clrs,
         border_color = NA,
         main = "Root",
         fontsize = 15
         )

x2 <- get_list_of_DEGs(plantpart = "shoot", method = "treatment", log2FC_threshold = 0) %>% rownames_to_column("genes")
y2 <- filter_counts_based_on_compartment(comprtmt = "shoot") %>% t() %>% as.data.frame() %>% rownames_to_column("genes")
DEG_counts2 <- left_join(x2, y2, by = "genes")
DEG_counts2 <- DEG_counts2[, -c(2,3,4,5,6)]
DEG_counts2 <- DEG_counts2 %>% column_to_rownames("genes")

sip2 <- data.frame(names(DEG_counts2), treatment) %>% column_to_rownames("names.DEG_counts2.")

pheatmap(DEG_counts2,
         gaps_col = 5,
         show_rownames = FALSE,
         show_colnames = FALSE,
         cluster_rows = TRUE,                      
         cluster_cols = TRUE,
         scale = "row",
         col = myCol,
         treeheight_row = 0,
         cutree_rows = 1,
         annotation_col = sip2,
         annotation_colors = clrs,
         border_color = NA,
         main = "Shoot",
         fontsize = 15
         )

```

Indeed the samples from the same treatment cluster together.

## LRT

Compare Wald test with LRT test, which checks the null-model with the full model.
First for the complete tomato dataset.
```{r message=FALSE, warning=FALSE}
source("Sly1/scripts/compare_Wald_vs_LRT.R")
compare_wald_vs_LRT(compartment = c("root","shoot"), method = "compartment+treatment")
```

This is a confirmation that there are no overlapping DEGs between compartments induced by solA treatment.
Check also the within compartment comparisons:
```{r message=FALSE, warning=FALSE}
compare_wald_vs_LRT(compartment = "root", method = "treatment")

compare_wald_vs_LRT(compartment = "shoot", method = "treatment")
```

For the root, LRT gives less (17) DEGs than Wald (22), but all of the LRT genes overlap with Wald. For the shoot, LRT gives 23 DEGs and Wald 46, again all of the LRT DEGs overlap with Wald. This is a reasonable confirmation of our DEGs.

# Annotation of DEGs

To infer the function of solA on tomato, we have to infer the function of the DEGs.
Since looking at the complete dataset hardly gives us DEGs, we only look at yes/no solA within the compartments.
```{r message=FALSE, warning=FALSE}
source("Sly1/scripts/get_annotated_DEGs.R")
DEGsboth <- get_annotated_DEGs(plantpart = c("root", "shoot"), method = "compartment+treatment", name = "Sly1/output/annotated_DEGs_both.csv")
DEGroot <- get_annotated_DEGs(plantpart = "root", method = "treatment", name = "Sly1/output/annotated_DEGs_root.csv")
DEGshoot <- get_annotated_DEGs(plantpart = "shoot", method = "treatment", name = "Sly1/output/annotated_DEGs_shoot.csv")
DEGboth$compartment <- rep("both", count(DEGboth))
DEGroot$compartment <- rep("root", count(DEGroot))
DEGshoot$compartment <- rep("shoot", count(DEGshoot))
DEGSly <- rbind(DEGroot, DEGshoot, DEGboth)
DEGSly$original_genename <- substr(DEGSly$original_genename, start = 6, stop = 23)
uniquepaste <- function(x) { paste(unique(x), sep = ',', collapse = ",") }
DEGSly <- aggregate(DEGSly, by = list(DEGSly$original_genename, DEGSly$compartment), FUN = uniquepaste)
write_delim(DEGSly, "Sly1/output/allDEGs.csv", delim = ";")
```

# Individual gene plots

Make individual plots of interesting genes:

```{r geneplots, fig.height = 3, fig.width = 8, message = FALSE, warning = FALSE}
source("Sly1/scripts/plotcounts.R")

p1 <- plotcounts(gene = "mRNA:Solyc04g079450.4.1",dds3, "serpin 3")
p2 <- plotcounts(gene = "mRNA:Solyc01g102895.1.1",dds3, "GLP4a", "r")
p3 <- plotcounts(gene = "mRNA:Solyc01g102890.4.1",dds3, "GLP4b", "r")
p4 <- plotcounts(gene = "mRNA:Solyc07g055740.1.1",dds3, "STR-2", "r")
p5 <- plotcounts(gene = "mRNA:Solyc10g085870.1.1",dds3, "UGT", "r")
p7 <- plotcounts(gene = "mRNA:Solyc12g088170.2.1",dds3, "acyl transferase", "r")
p6 <- plotcounts(gene = "mRNA:Solyc09g091510.3.1",dds3, "CHS1", "r")
p23 <- p2 + p3
pA <- p1 + p23 + guide_area() + plot_layout(guides = "collect", ncol = 3) + plot_annotation(title = "Immunity-related expression in tomato")
p45 <- p4 + p5
p67 <- p6 + p7
pB <- p45 - p67 + guide_area() + plot_layout(guides = "collect", ncol = 3) + plot_annotation(title = "Biosynthesis of secondary metabolites-related expression in tomato")

p8 <- plotcounts(gene = "mRNA:Solyc07g049530.3.1",dds3, "ACO1", "s")
p9 <- plotcounts(gene = "mRNA:Solyc02g081190.5.1",dds3, "ACO4", "s")
p10 <- plotcounts(gene = "mRNA:Solyc08g007820.1.1",dds3, "ERF10-2", "s")
p11 <- plotcounts(gene = "mRNA:Solyc09g015020.1.1",dds3, "HSP17.7B", "s")
p12 <- plotcounts(gene = "mRNA:Solyc04g011440.4.1",dds3, "ERHSPC70", "s")

pC <- p8 + p9 + p10 + p11 + p12 + plot_layout(guides = "collect", ncol = 5) + plot_annotation(title = "Ethylene biosynthesis and signalling-related expression in tomato")

p13 <- plotcounts(gene = "mRNA:Solyc09g011080.3.1",dds3, "activase", "r")
p14 <- plotcounts(gene = "mRNA:Solyc10g018300.3.1",dds3, "TKL1", "r")
p15 <- plotcounts(gene = "mRNA:Solyc03g034220.3.1",dds3, "RuBP ss", "r")
p16 <- plotcounts(gene = "mRNA:Solyc03g062720.3.1",dds3, "photosynth. NDH subcom. B2", "s")

p1314 <- p13 + p14
p1516 <- p15 + p16

pD <- p1314 - p1516 + guide_area() + plot_layout(guides = "collect", ncol = 3) + plot_annotation(title = "Photosynthesis-related expression in tomato")

p17 <- plotcounts(gene = "mRNA:Solyc09g097770.3.1",dds3, "CWP", "r")
p18 <- plotcounts(gene = "mRNA:Solyc02g091920.3.1",dds3, "XTH7", "r")

pE <- (p17 + p18) - plot_spacer() + plot_spacer() + plot_layout(guides = "collect", ncol = 3) + plot_annotation(title = "Cell wall expansion-related expression in tomato")

pA
pB
pC
pD
pE

```


Plots of overlapping genes with Arabidopsis

```{r overlap_AthSly, warning = FALSE, message = FALSE}
p19 <- plotcounts(gene = "mRNA:Solyc09g015020.1.1",dds3,"SlyHSP17.4A/\nHSP18.1","s")
p20 <- plotcounts(gene = "mRNA:Solyc01g102890.4.1",dds3,"SlyGLP1M7","r")
p21 <- plotcounts(gene = "mRNA:Solyc10g085870.1.1",dds3,"SlyUGT73","r")
p22 <- plotcounts(gene = "mRNA:Solyc10g078590.2.1",dds3,"SlyJRL3","s")

p19 + p20 + p21 + p22 + plot_layout(guides = "collect", ncol = 4)
```

Plots of genes NOT DEG in Sly, but in all three treatments of Arabidopsis
```{r overlap_AthSly, warning = FALSE, message = FALSE}
p100 <- plotcounts(gene = "mRNA:Solyc05g052570.3.1",dds3, "SlySZF2")
p101 <- plotcounts(gene = "mRNA:Solyc01g095100.4.1",dds3, "WRKY22") + plotcounts(gene = "mRNA:Solyc10g011910.5.1",dds3, "WRKY22")
p102 <- plotcounts(gene = "mRNA:Solyc07g054080.3.1",dds3, "RMA1") + plotcounts(gene = "mRNA:Solyc10g008400.1.1",dds3, "RMA1") + plotcounts(gene = "mRNA:Solyc10g008410.1.1",dds3, "RMA1") + plotcounts(gene = "mRNA:Solyc12g010670.1.1",dds3, "RMA1")
```
